src/main/java/ch/njol/yggdrasil/DefaultYggdrasilInputStream.java

```diff
@@ -137,21 +124,17 @@ private int readInt() throws IOException {
 	}
 	
 	private int readUnsignedInt() throws IOException {
-		final int b = read();
+		int b = read();
 		if ((b & 0x80) != 0)
 			return (b & ~0x80) << 8 | read();
 		return b << 24 | read() << 16 | read() << 8 | read();
 	}
 	
 	private long readLong() throws IOException {
```

Comment:
This can be done more efficiently by reading out 8 raw bytes, checking if the final one is negative to throw the exception, and then assembling them. It would avoid the pointless checking of each, since we only care if *something* reads out of the file, not which one it is.